pinecone/core/grpc/index_grpc.py

```diff
@@ -77,6 +80,7 @@ def __init__(self, index_name: str, channel=None, grpc_config: GRPCClientConfig
             "service-name": index_name,
             "client-version": CLIENT_VERSION
         }
+        self.pool = ThreadPoolExecutor()
```

Comment:
It'd be nice to avoid managing another threadpool if we can. WDYT about reusing PineconeGrpcFuture like upsert/delete already do and doing the response parsing in the result() method? It's simpler, though it may hurt performance. But I'm not actually sure about the performance impact; if response parsing is quick enough relative to the network/backend latency maybe the extra thread switching will hurt more than it helps. It'd be good to get some rough performance numbers on that (one calling thread, response parsing on caller thread or in extra threadpool, compare throughputs/latencies).

Btw, the current code will default the ThreadPoolExecutor's max_workers to the number of processors, which isn't what we want (https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ThreadPoolExecutor). In contrast, the openapi client generated code defaults it to some constant multiplied by the number of processors (5\*numproc? 10\*numproc? I forget). If we end up going with the extra threadpool, we should fix this.

Reply:
I put in a thread pool on the assumption it would hurt performance otherwise, though I didn't go far enough to check actual numbers.

Also yeah, if we keep the threadpool, changing default workers would be good.